code/modules/food_and_drinks/food/foods/ingredients.dm

```diff
@@ -129,6 +129,50 @@
 	list_reagents = list("nutriment" = 1)
 
 
+///cookies by Ume
+
+/obj/item/reagent_containers/food/snacks/cookiedough
+	name = "cookie dough"
+	desc = "The base for tasty cookies."
+	icon_state = "cookiedough"
+	list_reagents = list("nutriment" = 5, "sugar" = 5)
+
+// Dough + rolling pin = flat cookie dough
+/obj/item/reagent_containers/food/snacks/cookiedough/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/kitchen/rollingpin))
+		if(isturf(loc))
+			new /obj/item/reagent_containers/food/snacks/cookiedough_flat(loc)
```

Comment:
Instead of doing it this way, why not have a `flat` var that determines how it acts? This saves you having to make a subtype and duplicating code, and is more object-oriented (and keeps object deletion and creation down which is a benefit I guess)

e.g. (psuedocode below)

```
/obj/item/reagent_containers/food/snacks/cookiedough/attackby(mob/user, obj/item/W)
    if(istype(W, rollingpin) && !flat)
        to_chat(user, "you flatten [src].")
        flat = TRUE
    else if(istype(W, cookie_cutter) && flat)
        to_chat(user, "you cut [src] into cookie shapes.")
        new obj/item/food/raw_cookie(loc)
        qdel(src)
    else
        ..()
```

Reply:
Here I just copied the normal dough system (You know, rolling pin, knife) but it's a great idea too. I'll add that code and test if it works properly (As I see it, it should), and credit you if you don't mind

Reply:
I don't generally care about being credited for minor things like this, if you want to go ahead though 👍 

Also, while copy-pasting is fine for some things, I'd try to keep that to a minimum, as we have so much substandard code copy-pasting it may be detrimental to how good your overall code is. Just my two cents 😄

Reply:
Also fixed this